de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/service/impl/BPartnerDAO.java

```diff
@@ -1037,18 +1038,39 @@ public I_C_BPartner_Location retrieveBPartnerLocation(@NonNull final BPartnerLoc
 				.addOnlyActiveRecordsFilter();
 
 		queryBuilder.orderBy()
-				.addColumn(I_C_BP_Relation.COLUMNNAME_C_BP_Relation_ID);
+			.addColumn(I_C_BP_Relation.COLUMNNAME_C_BP_Relation_ID);
 
-		final I_C_BP_Relation billtoRelation = bpRelationQueryBuilder
+		final Stream<BPRelation> relationsStream = bpRelationQueryBuilder
 				.create()
-				.firstOnly(I_C_BP_Relation.class); // just added an UC
+				.stream()
+				.map(bpRelationRecord -> ofRelationRecord(bpRelationRecord));
 
-		if (billtoRelation != null)
+		Optional<BPartnerLocationId> relBPLocationId = Optional.empty();
+		if (query.getRelationBPartnerLocationId() != null)
```

Comment:
AFAIU, if getRelationBPartnerLocationId is null, then the whole `isAlsoTryRelation`-thing is discarded.
So we actually don't even need an extract `isAlsoTryRelation` field in the query.

=> Make sure to 
* either have this method work with all combinations of the query (like `isAlsoTryRelation=true` and `relationBPartnerLocationId=null`- feel free to write unit tests)
  * note that this is assumed in the javadoc i added to `BPartnerLocationQuery`
* oor split it into two public methods that do the different things which `OrderBL` and `InvoiceToExportFactory` need.
  * pls then make sure that my javadoc i added to `BPartnerLocationQuery` is updated too - if any javadoc is still needed